extensions/blocks/story/story.php

```diff
@@ -84,17 +163,97 @@ class="wp-story-video intrinsic-ignore wp-video-%3$s"
  *
  * @return string
  */
-function render_slide( $media, $index ) {
+function render_slide( $media, $index = 0 ) {
+	$media_template = '';
+	switch ( ! empty( $media['type'] ) && $media['type'] ) {
+		case 'image':
+			$media_template = render_image( $media, $index );
+			break;
+		case 'video':
+			$media_template = render_video( $media, $index );
+			break;
+	}
 	return sprintf(
-		'<li class="wp-story-slide" style="display: %s;">
+		'<div class="wp-story-slide" style="display: %s;">
 			<figure>
 				%s
 			</figure>
-		</li>',
+		</div>',
 		0 === $index ? 'block' : 'none',
-		'image' === $media['type']
-			? render_image( $media, $index )
-			: render_video( $media, $index )
+		$media_template
+	);
+}
+
+/**
+ * Render the top right icon on top of the story embed
+ *
+ * @param array $settings The block settings.
+ *
+ * @return string
+ */
+function render_top_right_icon( $settings ) {
+	$show_slide_count = isset( $settings['showSlideCount'] ) ? $settings['showSlideCount'] : false;
+	$slide_count      = isset( $settings['slides'] ) ? count( $settings['slides'] ) : 0;
+	if ( $show_slide_count ) {
+		// Render the story block icon along with the slide count.
+		return sprintf(
+			'<div class="wp-story-embed-icon">
+				<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" role="img" aria-hidden="true" focusable="false">
+					<path d="M0 0h24v24H0z" fill="none"></path>
+					<path fill-rule="evenodd" clip-rule="evenodd" d="M6 3H14V17H6L6 3ZM4 3C4 1.89543 4.89543 1 6 1H14C15.1046 1 16 1.89543 16 3V17C16 18.1046 15.1046 19 14 19H6C4.89543 19 4 18.1046 4 17V3ZM18 5C19.1046 5 20 5.89543 20 7V21C20 22.1046 19.1046 23 18 23H10C8.89543 23 8 22.1046 8 21H18V5Z"></path>
+				</svg>
+				<span>%s</span>
+			</div>',
+			$slide_count
+		);
+	} else {
+		// Render the Fullscreen Gridicon.
+		return (
+			'<div class="wp-story-embed-icon-expand">
+				<svg class="gridicon gridicons-fullscreen" height="24" width="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
+					<g>
+						<path d="M21 3v6h-2V6.41l-3.29 3.3-1.42-1.42L17.59 5H15V3zM3 3v6h2V6.41l3.29 3.3 1.42-1.42L6.41 5H9V3zm18 18v-6h-2v2.59l-3.29-3.29-1.41 1.41L17.59 19H15v2zM9 21v-2H6.41l3.29-3.29-1.41-1.42L5 17.59V15H3v6z"></path>
+					</g>
+				</svg>
+			</div>'
+		);
+	}
+}
+
+/**
+ * Render a pagination bullet
+ *
+ * @param array $slide_index The slide index it corresponds to.
+ *
+ * @return string
+ */
+function render_pagination_bullet( $slide_index ) {
+	return sprintf(
+		'<button class="wp-story-pagination-bullet" aria-label="Go to slide %d">
```

Comment:
I didn't know, do you know more about this?
In general, I try to stay away from anchor tags when there is no `href` attribute to point to, which is the case here

Reply:
This may tell you a bit more:
- p5TWut-gm-p2
- https://github.com/Automattic/wp-calypso/pull/31175 (that's back when Jetpack blocks were still developed in the Calypso monorepo, but the code still uses `a` tags to this day:
https://github.com/Automattic/jetpack/blob/3b5cf194ac4e43c57f07784d6f0171802314d3d9/extensions/blocks/slideshow/slideshow.js#L169

As you can see, the no href issue was discussed in that PR as well.

Reply:
Interesting, will convert every button, just to be on the safe side